File: benchmarks/poly/jsondeserialize/jsondeserialize.rb
def main
  _t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond)
  n = 400000
  checksum = 0
  total_name_len = 0

  n.times do |i|
    line = "{\"id\":#{i},\"name\":\"user#{i}\",\"score\":#{i % 100}}"

    id_val, name_val, score_val =
      line.match(/"id":(\d+),"name":"(user\d+)","score":(\d+)}/).captures

    checksum = (checksum * 31 + id_val.to_i + score_val.to_i) % 1000000007
    total_name_len += name_val.length
  end

  puts checksum
  puts total_name_len
  $stderr.puts "bench_ns=#{Process.clock_gettime(Process::CLOCK_MONOTONIC, :nanosecond) - _t0}"
end

main

File: benchmarks/poly/jsondeserialize/jsondeserialize.pl
use strict;
use warnings;
use Time::HiRes qw(clock_gettime CLOCK_MONOTONIC);

sub main {
    my $_t0 = clock_gettime(CLOCK_MONOTONIC);
    my $n = 400000;
    my $checksum = 0;
    my $total_name_len = 0;

    for (my $i = 0; $i < $n; $i++) {
        my $score = $i % 100;
        my $line = qq({"id":$i,"name":"user$i","score":$score});

        my ($id_val, $name_val, $score_val) =
            $line =~ /"id":(\d+),"name":"(user\d+)","score":(\d+)}/;

        $checksum = ($checksum * 31 + $id_val + $score_val) % 1000000007;
        $total_name_len += length($name_val);
    }

    print "$checksum\n";
    print "$total_name_len\n";
    printf STDERR "bench_ns=%.0f\n", (clock_gettime(CLOCK_MONOTONIC) - $_t0) * 1e9;
}

main();
